cpp/log/cluster_state_controller_test.cc

```diff
@@ -501,6 +501,16 @@ TEST_F(ClusterStateControllerTest, TestGetLocalNodeState) {
 
 
 TEST_F(ClusterStateControllerTest, TestNodeHostPort) {
+  // Allow some time for our view of the node state to stabilize and then
+  // check that it has. Ideally we would wait for this to definitely happen
+  // but there seems to be no easy way to arrange this.
+  sleep(1);
+  const ClusterNodeState orig_node_state(GetNodeStateView(kNodeId1));
+  EXPECT_EQ(kNodeId1, orig_node_state.hostname());
+  EXPECT_EQ(9001, orig_node_state.log_port());
```

Comment:
Maybe those two should be `ASSERT_EQ` instead? As in, if these fails, the rest of the test is kind of meaningless, right?

I guess you could make it a little loop like this, to try a bit harder, but not *too* hard either?

```C++
for (int i = 0; i < 10; ++i) {
  const ClusterNodeState orig_node_state(GetNodeStateView(kNodeId1));
  if (orig_node_state.hostname() == kNodeId1 && orig_node_state.log_port() == 9091) {
    break;
  }
  sleep(1);
}
```